syntax ="proto3";

message crearUsuarioRequest {
    string username = 1;
    string email = 2;
    string password = 3;
    string tipo = 4;
}

message crearUsuarioResponse {
    string username = 1;
    string mensaje = 2;
    /*
        mensajes:
            "Alta Exitosa" (cuando no existe usuario en BD y Se puede crear)
            "Alta Fail"
            "Usuario Existente" (cuando ya existe el usuario en BD)
    */
}

message loguearUsuarioRequest{
    string username = 1;
    string password = 3;
    
}

/*
username = contiene el nombre de usuario
estado:
    -"Log Ok"
    -"Contrase√±a erronea"
    -"Usuario inexistente"
*/
message loguearUsuarioResponse{
    string username = 1;
    string estado = 2;
}


message solicitudDeSeguidorRequest{
    string usuarioQueSigue = 1;
    string usuarioSeguido = 2;

}

message solicitudDeSeguidorResponse{
    string mensaje = 1;
    /*
        mensajes:
            "Estas siguiendo a este usuario"
            "No se puede seguir a este usuario"
    */
}

message agregarRecetaAFavoritosRequest{
    string usuario = 1;
    string idReceta = 2;

}

message agregarRecetaAFavoritosResponse{
    string mensaje = 1;
    /*
        mensajes:
            "Agregado a favoritos"
            "Error"
    */
}

message traerUsuariosQueSigoRequest{
    string usuario= 1;

}

message traerUsuariosQueSigoResponse{
    repeated string usuarios = 1;
}

service servicioUsuario{
    rpc crearUsuario (crearUsuarioRequest) returns (crearUsuarioResponse);
    rpc loguearUsuario (crearUsuarioRequest) returns (crearUsuarioResponse);
    rpc seguirUsuario(solicitudDeSeguidorRequest) returns (solicitudDeSeguidorResponse);
    rpc agregarRecetaAFavoritos(agregarRecetaAFavoritosRequest) returns (agregarRecetaAFavoritosResponse);
    rpc traerUsuariosQueSigo(traerUsuariosQueSigoRequest) returns (traerUsuariosQueSigoResponse);
}